packit_service/models.py

```diff
@@ -76,8 +76,7 @@ def get_sa_session() -> Session:
     global session_instance
```

Comment:
Due to this line I would say we'll still have a global session shared between threads, so it won't be thread local :)

Reply:
How about storing the session in [Flask.g](https://flask.palletsprojects.com/en/1.1.x/appcontext/#storing-data)?

Reply:
I dont understand why this needs to be global but it fixed that issue, i tested it multiple times.

Reply:
it should not be global, scoped session is already handling it - it will always return same session in same thread, regardless how many times it is called.

https://docs.sqlalchemy.org/en/13/orm/contextual.html

Reply:
also some useful FAQ
https://docs.sqlalchemy.org/en/13/orm/session_basics.html#session-faq-whentocreate

Reply:
>  scoped session is already handling it - it will always return same session in same thread, regardless how many times it is called.

thanks

Reply:
on removing the global declaration, I get 

is not bound to a Session; attribute refresh operation cannot proceed (Background on this error at: http://sqlalche.me/e/bhk3),

Reply:
Can we have the following outside `get_sa_sesssion()`?

```python
engine = create_engine(get_pg_url())
Session = scoped_session(sessionmaker(bind=engine))
```

(not sure if this would matter...)